app/models/user.rb

```diff
@@ -97,4 +97,11 @@ def only_person_of_team_missing_fab?
     upcoming_fab_still_missing? and !upcoming_fab_still_missing_for_team_mate?
   end
 
+  def username
+    email.split("@").first
+  end
+
+  def last_whereabouts
+    where_messages.order('sent_at DESC').first
```

Comment:
Non-blocking, just fyi it's also possible to get Rails to hook this up as an association:

```ruby
  has_one :last_whereabouts, class_name: "WhereMessage", ->{ order(sent_at: :desc) }
```

Reply:
neat!